src/structure_managers/adaptor_neighbour_list.hh

```diff
@@ -300,11 +304,11 @@ namespace rascal {
     //! get dimension dependent neighbour indices (surrounding cell and the cell
     //! itself
     template <size_t Dim, class Container_t>
-    std::vector<int>
-    get_neighbours_atom_tag(const int current_atom_tag,
-                            const std::array<int, Dim> & ccoord,
-                            const Container_t & boxes) {
-      std::vector<int> neighbours_atom_tag{};
+    void get_neighbours_atom_tag(const int current_atom_tag,
```

Comment:
the reference is modified so no need to return. it avoids creating intermediate variables neighbours_atom_tag over and over again

Reply:
The function name is a bit misleading but modifying by reference is a very valid c/c++ construct since it is the easiest way to modify on several variables while one can only return one.
The clear does not free memory but only reset the size of the array so it effectively does not allocate new memory.
RVO might be able to optimize out this but I checked with the profiler and this change corresponded to some performance gain.

Reply:
>  the change breaks compatibility with code that expected the old behaviour

it is used here only so no code expects the old behaviour.